components/movementsensor/adxl345/adxl345.go

```diff
@@ -189,10 +235,90 @@ func NewAdxl345(
 			}
 		}
 	})
-
+	sensor.interruptsFound = make(map[string]int)
+	sensor.readInterrupts(sensor.cancelContext)
+	err = sensor.configureInterruptRegisters(ctx)
+	if err != nil {
+		sensor.cancelFunc()
```

Comment:
Mild preference for putting in a comment explaining that we need to do this to shut down the goroutine that reads the sensor in the background.